resources/app.py

```diff
@@ -385,6 +491,322 @@ def delete_bed_reservation(patient_id: str, db=Depends(get_db)):
+    db=Depends(get_db),
+):
+    existing_employee_schedule = (
+        db.query(employee_schedule_model)
+        .filter(employee_schedule_model.ID_entry == schedule_id)
+        .first()
+    )
+    if existing_employee_schedule is None:
+        raise HTTPException(status_code=404, detail="Employee schedule not found")
+
+    try:
+        db.query(employee_schedule_model).filter(
+            employee_schedule_model.ID_entry == schedule_id
+        ).update(employee_schedule.model_dump())
+        db.commit()
+    except Exception:
+        raise HTTPException(
+            status_code=400, detail="Error while updating employee schedule."
+        )
+    return {"message": "Employee schedule updated successfully."}
+
+
+@app.delete("/delete/employee_schedule/{schedule_id}", tags=["Employee Schedule"])
+def delete_employee_schedule(schedule_id: int, db=Depends(get_db)):
+    employee_schedule = (
+        db.query(employee_schedule_model)
+        .filter(employee_schedule_model.ID_entry == schedule_id)
+        .first()
+    )
+    if employee_schedule is None:
+        raise HTTPException(status_code=404, detail="Employee schedule not found")
+    try:
+        db.delete(employee_schedule)
+        db.commit()
+    except Exception:
+        raise HTTPException(
+            status_code=400, detail="Error while deleting employee schedule."
+        )
+    return {"message": "Employee schedule deleted successfully."}
+
+
+# ======================== Medical Procedure Management ========================
+@app.get("/get/medical_procedure/all", tags=["Medical Procedure"])
+def get_medical_procedures(db=Depends(get_db)):
+    medical_procedures = db.query(medical_procedure_model).all()
+    if not medical_procedures:
+        raise HTTPException(status_code=404, detail="Medical procedures not found")
+    return medical_procedures
+
+
+@app.get("/get/medical_procedure/id/{procedure_id}", tags=["Medical Procedure"])
+def get_medical_procedure(procedure_id: int, db=Depends(get_db)):
+    medical_procedure = (
+        db.query(medical_procedure_model)
+        .filter(medical_procedure_model.ID_procedure == procedure_id)
+        .first()
+    )
+    if medical_procedure is None:
+        raise HTTPException(status_code=404, detail="Medical procedure not found")
+    return medical_procedure
+
+
+@app.post("/create/medical_procedure", tags=["Medical Procedure"])
+def create_medical_procedure(
+    medical_procedure: create_medical_procedure_schema, db=Depends(get_db)
+):
+    medical_procedure = medical_procedure_model(**medical_procedure.model_dump())
+    try:
+        db.add(medical_procedure)
+        db.commit()
+    except Exception:
+        raise HTTPException(
+            status_code=400, detail="Error while adding medical procedure."
+        )
+    return {"message": "Medical procedure added successfully."}
+
+
+@app.patch("/update/medical_procedure/{procedure_id}", tags=["Medical Procedure"])
+def update_medical_procedure(
```

Comment:
i think we should make all fields optional, now we have to provide all fields to not finish with 422.

For now it is rather a put request, as we need to edit every field to change

As we want to enable partial update, please do it similar to: https://github.com/BoredOtter/hms/blob/master/pharmacy/app.py#L89